arangod/Replication2/ReplicatedLog/ReplicatedLogMetrics.cpp

```diff
@@ -0,0 +1,107 @@
+    : ReplicatedLogMetrics(&metricsFeature) {}
+
+template <typename Builder, bool mock>
+auto ReplicatedLogMetrics::createMetric(arangodb::MetricsFeature* metricsFeature)
+    -> std::shared_ptr<typename Builder::metric_t> {
+  TRI_ASSERT((metricsFeature == nullptr) == mock);
+  if constexpr (!mock) {
+    return metricsFeature->addShared(Builder{});
+  } else {
+    return std::dynamic_pointer_cast<typename Builder::metric_t>(Builder{}.build());
+  }
+}
+
+template <typename MFP, std::enable_if_t<std::is_same_v<arangodb::MetricsFeature*, MFP> || std::is_null_pointer_v<MFP>, int>, bool mock>
+ReplicatedLogMetrics::ReplicatedLogMetrics(MFP metricsFeature)
+    : replicatedLogNumber(
+          createMetric<arangodb_replication2_replicated_log_number, mock>(metricsFeature)),
+      replicatedLogAppendEntriesRttUs(
+          createMetric<arangodb_replication2_replicated_log_append_entries_rtt, mock>(
+              metricsFeature)),
+      replicatedLogFollowerAppendEntriesRtUs(
+          createMetric<arangodb_replication2_replicated_log_follower_append_entries_rt, mock>(
+              metricsFeature)),
+      replicatedLogCreationNumber(
+          createMetric<arangodb_replication2_replicated_log_creation_total, mock>(
+              metricsFeature)),
+      replicatedLogDeletionNumber(
+          createMetric<arangodb_replication2_replicated_log_deletion_total, mock>(
+              metricsFeature)),
+      replicatedLogLeaderNumber(
+          createMetric<arangodb_replication2_replicated_log_leader_number, mock>(
+              metricsFeature)),
+      replicatedLogFollowerNumber(
+          createMetric<arangodb_replication2_replicated_log_follower_number, mock>(
+              metricsFeature)),
+      replicatedLogInactiveNumber(
+          createMetric<arangodb_replication2_replicated_log_inactive_number, mock>(
+              metricsFeature)),
+      replicatedLogLeaderTookOverNumber(
+          createMetric<arangodb_replication2_replicated_log_leader_took_over_total, mock>(
+              metricsFeature)),
+      replicatedLogStartedFollowingNumber(
+          createMetric<arangodb_replication2_replicated_log_started_following_total, mock>(
+              metricsFeature)),
+      replicatedLogInsertsBytes(
+          createMetric<arangodb_replication2_replicated_log_inserts_bytes, mock>(
+              metricsFeature)),
+      replicatedLogInsertsRtt(
+          createMetric<arangodb_replication2_replicated_log_inserts_rtt, mock>(
+              metricsFeature)) {
+#ifndef ARANGODB_USE_GOOGLE_TESTS
+  static_assert(!mock);
+  static_assert(!std::is_null_pointer_v<MFP>);
+#endif
+}
+
+MeasureTimeGuard::MeasureTimeGuard(std::shared_ptr<Histogram<log_scale_t<std::uint64_t>>> histogram) noexcept
+    : _start(std::chrono::steady_clock::now()), _histogram(std::move(histogram)) {}
+
+void MeasureTimeGuard::fire() {
+  auto const endTime = std::chrono::steady_clock::now();
```

Comment:
It does not seem to be worth to calculate the `endTime` if `_histogram` is false, so we should probably move this line into the `if (_histogram)` block below.

Reply:
Done